base/server/systemd/SystemdStartupNotifier.java

```diff
@@ -0,0 +1,59 @@
+// --- BEGIN COPYRIGHT BLOCK ---
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; version 2 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// (C) 2020 Red Hat, Inc.
+// All rights reserved.
+// --- END COPYRIGHT BLOCK ---
+
+package com.netscape.cmscore.systemd;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
```

Comment:
I'm not sure... from my POV the jna package is supported by Red Hat so it should work on the JDK supported by Red Hat, and that is good enough for us.  Otherwise we can switch it to JNI.

Reply:
I mean, this is fine, but JSS is already a JNI'd library. We in PKI already use JNI (see com.netscape.symkey). I'm less liking adding _two_ different native bindings to the same project than I am JNA. :-)

Reply:
*Reductio ad absurdum*: We use JNI, so we must continue to use JNI only (or convert everything to another FFI system all at once).

I would sooner convert this PR to exec the `systemd-notify(1)` program, than convert it to JNI.  At the start of this week I had never even heard of JNA, but now that I have seen it, I never want to use JNI ever again :D

Reply:
My concern about JNA is the package maintenance. The package is available now on Fedora & RHEL, but there seems to be an effort to move it into Eclipse module on Fedora, and we might be required to add it into pki-deps module on RHEL in the future. Unfortunately this is the current state of Java on Fedora/RHEL. Once we move to Maven we'll have more flexibility on what libraries to use.

Reply:
@edewata we can switch from JNA to JNI or to exec'ing `systemd-notify(1)` in the future, if required.

Reply:
In the past ``systemd-notify`` was not 100% stable. systemd handles notifications asynchrony. It is possible that the program exits before systemd is able to retrieve its cgroups.

Reply:
OK, yeah, we can use JNA now and consider the alternative if it becomes a problem in the future.